packages/common-cosmos/src/codegen/codegen-controller.ts

```diff
@@ -31,6 +47,105 @@ export function isProtoPath(filePath: string, projectPath: string): boolean {
   return !!path.join(projectPath, filePath).startsWith(path.join(projectPath, './proto/'));
 }
 
+export function prepareCosmwasmJobs(
+  sortedAssets: Map<string, string>,
+  loadReadAbi: (filePath: string) => IDLObject,
+  upperFirst: (input?: string) => string
+): CosmwasmRenderJobType[] {
+  return Array.from(sortedAssets.keys()).map((key) => {
+    const value = sortedAssets.get(key);
+    const readContract = loadReadAbi(value);
+
+    const msgObjs: Record<string, string | null> = {
+      MsgInstantiateContract: readContract.instantiate?.title ?? null,
+      MsgMigrateContract: readContract.migrate?.title ?? null,
+      MsgExecuteContract: readContract.execute?.title ?? null,
+    };
+
+    const messages = Object.entries(msgObjs)
+      .filter(([_, value]) => value !== null)
+      .map(([msgType, msgTitle]) => ({
+        [upperFirst(msgType)]: upperFirst(msgTitle),
+      }));
+
+    return {
+      contract: upperFirst(key),
+      messages,
+    };
+  });
+}
+
+export function prepareSortedAssets(
+  datasources: SubqlCosmosRuntimeDatasource[],
+  projectPath: string
+): Map<string, string> {
```

Comment:
Whats the reasoning for using a map? I think code would be simpler with a Record